src/main/java/org/stellar/sdk/Transaction.java

```diff
@@ -92,6 +94,37 @@ public long getFee() {
     return mOperations;
   }
 
+  /**
+   * Returns returns the claimable balance ID for the CreateClaimableBalanceOperation at the given index within the transaction.
+   */
+  public String getClaimableBalanceId(int index) throws IOException {
+    if (index >= mOperations.length) {
+      throw new IllegalArgumentException("index: " + index + " is outside the bounds of the operations within this transaction");
+    }
+    if (!(mOperations[index] instanceof CreateClaimableBalanceOperation)) {
+      throw new IllegalArgumentException("operation at index " + index + " is not of type CreateClaimableBalanceOperation: " + mOperations[index].getClass());
+    }
+
+    OperationID id = new OperationID();
+    id.setDiscriminant(EnvelopeType.ENVELOPE_TYPE_OP_ID);
+    OperationID.OperationIDId body = new OperationID.OperationIDId();
+    body.setOpNum(new Uint32(index));
+    body.setSeqNum(new SequenceNumber(new Int64(getSequenceNumber())));
+    body.setSourceAccount(AccountConverter.disableMuxed().encode(getSourceAccount()));
+    id.setId(body);
+
+    ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
+    id.encode(new XdrDataOutputStream(outputStream));
+
+    ClaimableBalanceID result = new ClaimableBalanceID();
+    result.setDiscriminant(ClaimableBalanceIDType.CLAIMABLE_BALANCE_ID_TYPE_V0);
+    result.setV0(new Hash(Util.hash(outputStream.toByteArray())));
```

Comment:
This is probably obvious with context & usage but there's no risk of the hash somehow... not being sha2 at some point?

Reply:
unless we decide to to swap out sha2 for a different algorithm for transaction hashing it should remain as sha2